src/link.js

```diff
@@ -106,13 +102,12 @@ module.exports = function link(config, args, callback) {
     })
     .filter(dependency => dependency);
 
-  const tasks = dependencies.map((dependency) => (next) =>
-    async.waterfall([
-      dependency.config.commands.prelink || commandStub,
-      makeLink(project, dependency),
-      dependency.config.commands.postlink || commandStub,
-    ], next)
-  );
+  const tasks = dependencies.map(dependency => () => {
+    const pre = dependency.config.commands.prelink || commandStub;
+    const post = dependency.config.commands.postlink || commandStub;
```

Comment:
@Kureev I think we need to update these in `rnpm` to return a promise and bump `minor`